modules/videoio/test/test_precomp.hpp

```diff
@@ -60,7 +60,8 @@ inline std::string fourccToString(int fourcc)
 inline std::string fourccToStringSafe(int fourcc)
 {
     std::string res = fourccToString(fourcc);
-    std::replace_if(res.begin(), res.end(), [](uint8_t c){ return c < '0' || c > 'z'; }, '_');
+    // TODO: return hex values for invalid characters
+    std::transform(res.begin(), res.end(), res.begin(), [](uint8_t c) { return (c >= '0' && c <= 'z') ? c : (c == ' ' ? '_' : 'x'); });
```

Comment:
There is build regression with MSVS2019/2017: https://pullrequest.opencv.org/buildbot/builders/4_x-win64-vc16/builds/100191

Reply:
Related CI issue: https://github.com/opencv/ci-gha-workflow/issues/108

Reply:
Created PR #24087